addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.airvisualnode.internal.discovery;
+
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.airvisualnode.AirVisualNodeBindingConstants;
+import org.openhab.binding.airvisualnode.internal.config.AirVisualNodeConfig;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jcifs.netbios.NbtAddress;
+import jcifs.smb.SmbFile;
+
+/**
+ * Autodiscovery for AirVisual Node by searching for a host advertised with the NetBIOS name 'AVISUAL-<SerialNumber>'.
+ *
+ * @author Victor Antonovich - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true)
+public class AirVisualNodeDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(AirVisualNodeDiscoveryService.class);
+
+    public static final String AVISUAL_WORKGROUP_NAME = "MSHOME";
+
+    private static final Pattern AVISUAL_NAME_PATTERN = Pattern.compile("^AVISUAL-([^/]+)$");
+
+    private final Runnable scannerRunnable;
+
+    private ScheduledFuture<?> backgroundDiscoveryFuture;
+
+    private DiscoveryServiceCallback discoveryServiceCallback;
+
+    public AirVisualNodeDiscoveryService() {
+        super(Collections.singleton(AirVisualNodeBindingConstants.THING_TYPE_AVNODE), 600, true);
+        scannerRunnable = createScannerRunnable();
+    }
+
+    @Override
+    public void setDiscoveryServiceCallback(DiscoveryServiceCallback discoveryServiceCallback) {
+        this.discoveryServiceCallback = discoveryServiceCallback;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting scan");
+        scheduler.execute(scannerRunnable);
```

Comment:
Maybe rewrite this as: `scheduler.execute(this::createScannerRunnable());` and remove the return part from `createScannerRunnable()`? (Although then the method name probably should be something more related to the running itself instead of creating the runnable`)